soa/service/s3.cc

```diff
@@ -412,38 +432,28 @@ performSync() const
                 message += (string("body (") + to_string(responseBody.size())
                             + " bytes):\n" + responseBody + "\n");
             }
+            if (!xmlError.first.empty()) {
+                message += ("S3 REST error code: "
+                            + xmlError.first
+                            + "; message: "
+                            + xmlError.second
+                            +"\n");
+            }
 
+            return message;
+        };
 
-            /* log so-called "REST error"
-               (http://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html)
-            */
-            if (responseHeaders.find("Content-Type: application/xml")
-                != string::npos) {
-                unique_ptr<tinyxml2::XMLDocument> localXml(
-                    new tinyxml2::XMLDocument()
-                );
-                localXml->Parse(responseBody.c_str());
-                auto element
-                    = tinyxml2::XMLHandle(*localXml).FirstChildElement("Error")
-                    .ToElement();
-                if (element) {
-                    message += ("S3 REST error: ["
-                                + extract<string>(element, "Code")
-                                + "] message ["
-                                + extract<string>(element, "Message")
-                                +"]\n");
-                }
-            }
+        if (!xmlError.first.empty()
```

Comment:
This is unlikely but we would now interpret a valid download of an xml file that contains a field error as an error.